spock-specs/src/test/groovy/org/spockframework/smoke/extension/ParallelSpec.groovy

```diff
@@ -0,0 +1,34 @@
+package org.spockframework.smoke.extension
+
+import org.spockframework.EmbeddedSpecification
+
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.atomic.AtomicInteger
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS
+
+class ParallelSpec extends EmbeddedSpecification {
+
+  static void incrementBlockAndCheck(AtomicInteger sharedResource, CountDownLatch countDownLatch)
+    throws InterruptedException {
+    int value = incrementAndBlock(sharedResource, countDownLatch)
+    assert value == sharedResource.get()
+  }
+
+  static int incrementAndBlock(AtomicInteger sharedResource, CountDownLatch countDownLatch)
+    throws InterruptedException {
+    int value = sharedResource.incrementAndGet()
+    countDownLatch.countDown()
+    countDownLatch.await(100, MILLISECONDS)
+    return value
+  }
+
+  static void storeAndBlockAndCheck(AtomicInteger sharedResource, CountDownLatch countDownLatch)
```

Comment:
Is it already used anywhere? Or, as Marc conjectures, it will be used for something?